### PR TITLE
BAEL-8898: Embedded MaraiDB using MariaDB4j

### PR DESCRIPTION
**Pull request by sazzer**

### CODE REVIEW

File: libraries-data-mariadb4j/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            <version>${mariadb.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <plugins>
+        </plugins>
+    </build>
```
Comment (theangrydev) [inline libraries-data-mariadb4j/pom.xml:34]: Remove empty section?

File: libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            db.stop();
+        }
+    }
+
+    @Test
+    void whenStartingADatabaseWithConfiguration_thenTheDatabaseIsUsable() throws Exception {
+        DBConfigurationBuilder config = DBConfigurationBuilder.newBuilder();
+        config.setPort(13306);
+        config.setDataDir("/tmp/data");
+        config.setBaseDir("/tmp/db");
```
Comment (theangrydev) [inline libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java:32]: Do not hard code a directory, use a unique temporary directory and clean it up after the test (or just don't use config.setDataDir and config.setBaseDir in this example)

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: libraries-data-mariadb4j/pom.xml (hunk 1 above)
Comment (sazzer) [reply libraries-data-mariadb4j/pom.xml]: It was originally there for the section on the maven plugin that got descoped. Removed the whole block now.

File: libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java (hunk 2 above)
Comment (sazzer) [reply libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java]: Dang. Good point :( Fixed.

File: libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java [hunk 3]
```diff
@@ -26,10 +28,12 @@ void whenStartingADatabase_thenTheDatabaseIsUsable() throws Exception {
 
     @Test
     void whenStartingADatabaseWithConfiguration_thenTheDatabaseIsUsable() throws Exception {
+        Path tempDir = Files.createTempDirectory("mariadb");
```
Comment (theangrydev) [inline libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java:31]: This temporary directory isn't cleaned up after the build - use something like `@TempDir` instead https://www.baeldung.com/junit-5-temporary-directory

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java (hunk 3 above)
Comment (sazzer) [reply libraries-data-mariadb4j/src/test/java/com/baeldung/data/mariadb4j/JavaLiveTest.java]: I didn't know about that one. Updated. :)

**Comment by theangrydev** [comment]
retest this please

**Review by theangrydev — APPROVED** [review]

**Pull request merged**